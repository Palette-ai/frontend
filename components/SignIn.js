import React, { useState } from 'react'
import { View, StyleSheet, Text, TextInput, TouchableOpacity, SafeAreaView, Button } from 'react-native'
import { useMutation } from '@apollo/client'

import { CREATE_USER } from '../queries/users'


function SignIn() {
	const [username, setUsername] = useState('')
	const [email, setEmail] = useState('')
	const [password, setPassword] = useState('')

	const [createUser] = useMutation(CREATE_USER, {
		onCompleted(data) {
			console.log(data);
		}
	})

	const onSubmit = () => {
		//e.preventDefault()
		console.log(username);
		createUser({
			variables: {
				record: { name: username, location: "Testy test test" }
			}
		})
	}

	return (
		<SafeAreaView>
			<View style={styles.container}>
				<Text>Or create a new account</Text>
				<TextInput
					name='username'
					value={username}
					onChangeText={text => setUsername(text)}
					placeholder='username'
				>
				</TextInput>
				<TextInput
					name='email'
					value={email}
					onChangeText={text => setEmail(text)}
					placeholder='email'
				>
				</TextInput>
				<TextInput
					name='password'
					value={password}
					onChangeText={text => setPassword(text)}
					placeholder='password'
				>
				</TextInput>
				<TouchableOpacity>
					<Button
						onPress={() => onSubmit(username)}
						title='Create new account'
					/>
				</TouchableOpacity>
			</View>
		</SafeAreaView>
	);
}

const styles = StyleSheet.create({
	container: {}
});

export default SignIn;