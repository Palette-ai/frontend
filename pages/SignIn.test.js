import React from 'react';
import SignIn from './SignIn';
import {shallow} from 'enzyme'
import "../setupTests"

describe('App renders correctly', () => {
  it('should render the App Component correctly', () => { 
    const app = shallow(<SignIn />);  
    expect(app).toMatchSnapshot();
  });
});
