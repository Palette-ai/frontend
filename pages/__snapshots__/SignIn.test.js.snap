// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly should render the App Component correctly 1`] = `
<View>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <ImageBackground
      source={1}
      style={
        Object {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
          "zIndex": -1,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "height": 300,
            "marginBottom": 80,
            "marginTop": -30,
            "top": -1,
            "transform": Array [
              Object {
                "rotate": "-15.00deg",
              },
            ],
            "width": 300,
            "zIndex": 1,
          }
        }
      />
      <LoginBox />
    </ImageBackground>
  </TouchableWithoutFeedback>
</View>
`;
